src/main/kotlin/com/github/quiltservertools/ledger/network/packet/handshake/HandshakeS2CPacket.kt

```diff
@@ -0,0 +1,33 @@
+package com.github.quiltservertools.ledger.network.packet.handshake
+
+import com.github.quiltservertools.ledger.network.packet.LedgerPacketTypes
+import net.minecraft.network.PacketByteBuf
+import net.minecraft.network.codec.PacketCodec
+import net.minecraft.network.packet.CustomPayload
+
+data class HandshakeS2CPacket(val content: HandshakeContent) : CustomPayload {
+    fun write(buf: PacketByteBuf?) {
+        // Ledger information
+        // Protocol Version
+        buf?.writeInt(content.protocolVersion)
+
+        // Ledger Version
+        buf?.writeString(content.ledgerVersion)
+
+        // We tell the client mod how many actions we are writing
+        buf?.writeInt(content.actions.size)
```

Comment:
I'm not sure, can you check if writeList exists by chance, i saw someone use it?

Reply:
The buffer provides `writeCollection`, but that uses `writeVarInt` instead of `writeInt` internally to send the collection size. I am pretty sure that would mean a difference in networking protocol.